src/modules/sensors/vehicle_angular_velocity/VehicleAngularVelocity.cpp

```diff
@@ -273,22 +282,32 @@ void VehicleAngularVelocity::Run()
 				perf_count_interval(_interval_perf, sensor_data.timestamp_sample);
 			}
 
+			// Guard against too small (< 0.2ms) and too large (> 20ms) dt's.
+			const float dt = math::constrain(((sensor_data.timestamp_sample - _timestamp_sample_prev) / 1e6f), 0.0002f, 0.02f);
+			_timestamp_sample_prev = sensor_data.timestamp_sample;
+
+
 			// get the sensor data and correct for thermal errors (apply offsets and scale)
 			const Vector3f val{sensor_data.x, sensor_data.y, sensor_data.z};
 
 			// apply offsets and scale
-			Vector3f rates{(val - _offset).emult(_scale)};
+			Vector3f angular_velocity_raw{(val - _offset).emult(_scale)};
 
 			// rotate corrected measurements from sensor to body frame
-			rates = _board_rotation * rates;
+			angular_velocity_raw = _board_rotation * angular_velocity_raw;
 
 			// correct for in-run bias errors
-			rates -= _bias;
+			angular_velocity_raw -= _bias;
 
-			// Filter: apply notch and then low-pass
-			CheckFilters(rates);
-			const Vector3f rates_filtered = _lowpass_filter.apply(_notch_filter.apply(rates));
+			// Differentiate angular velocity
+			const Vector3f angular_acceleration_raw = (angular_velocity_raw - _angular_velocity_prev) / dt;
+			_angular_velocity_prev = angular_velocity_raw;
+			_angular_acceleration_prev = angular_acceleration_raw;
 
+			// Filter: apply notch and then low-pass
+			CheckFilters();
+			const Vector3f angular_acceleration(_lp_filter_acceleration.apply(angular_acceleration_raw));
```

Comment:
I would suggest to apply a notch for the angular acceleration as well. Same notch frequency as for angular velocity.

Reply:
we can simplify further by differentiating the notch-ed angular velocity

Reply:
Done.